### PR TITLE
[MAINTENANCE] Update contributor instructions with additional steps that may be needed for installation to succeed.

### PR DESCRIPTION
**Pull request by alexsherstinsky**
### Scope
In the present author's experience, Ludwig installation has not gone smoothly (under `Anaconda` with Python 3.10.13) without additional steps and omissions.  This change recommends these steps and work-around procedures.

# Code Pull Requests

Please provide the following:

- a clear explanation of what your code does
- if applicable, a reference to an issue
- a reproducible test for your PR (code, config and data sample)

# Documentation Pull Requests

Note that the documentation HTML files are in `docs/` while the Markdown sources are in `mkdocs/docs`.

If you are proposing a modification to the documentation you should change only the Markdown files.

`api.md` is automatically generated from the docstrings in the code, so if you want to change something in that file, first modify `ludwig/api.py` docstring, then run `mkdocs/code_docs_autogen.py`, which will create `mkdocs/docs/api.md` .

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -51,6 +51,44 @@ Work on your self-assigned issue and eventually create a Pull Request.
    pre-commit install
    ```
 
+1. On MacOS with Apple Silicon, if this installation approach runs into errors, you may need to install the following prerequisites:
```
Comment (w4nderlust) [inline CONTRIBUTING.md:54]: all the lines have this 1. preceding them, not sure what the goal is, in theory in markdown that's an ordered list i beleive?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -51,6 +51,44 @@ Work on your self-assigned issue and eventually create a Pull Request.
...
+
+   ```bash
+   brew install cmake libomp
+   ```
+
+which require `homebrew` to be installed on your development machine.  Please install these libraries and then repeat the previous step:
+
+```bash
+pip install -e .
```
Comment (w4nderlust) [inline CONTRIBUTING.md:63]: this should probably be .[test]

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -51,6 +51,44 @@ Work on your self-assigned issue and eventually create a Pull Request.
...
+1. If the installation still does not complete successfully, then it is possible that the order of the
+   "requirements" files matters.  To carry out the installation in the proper sequence, please execute:
+
+   ```bash
+   pip install --use-pep517 --requirement requirements.txt --requirement requirements_benchmarking.txt \
+      --requirement requirements_distributed.txt --requirement requirements_explain.txt \
+      --requirement requirements_hyperopt.txt --requirement requirements_llm.txt \
+      --requirement requirements_serve.txt --requirement requirements_test.txt \
+      --requirement requirements_tree.txt --requirement requirements_viz.txt \
+      --requirement requirements_extra.txt --editable .
```
Comment (w4nderlust) [inline CONTRIBUTING.md:75]: not sure, but this should be equivalent to  pip install --use-pep517 -e .[test] (a parte fro mthe extras)
not sure what the pep directive changes, but i would verify before suggesting to manually change the requirement installation.

Also, in general, I suggest to checkout the setup.py script as it makes clear both the installation sub-packages and the order of the installation. if there are discoveries that come from reading it, I'd document them here too

File: CONTRIBUTING.md (hunk 1 above)
Comment (alexsherstinsky) [reply CONTRIBUTING.md:54]: @w4nderlust -- indeed I made a typo -- will fix as part of the next commit.  Thank you.

File: CONTRIBUTING.md (hunk 2 above)
Comment (alexsherstinsky) [reply CONTRIBUTING.md:63]: @w4nderlust Yes -- what was happening is that this command would not work in Z-shell; but now I figured it out -- the "[test]" has to be in quotes (after several examples provided by @tgaddair I finally realized that this is what was happening).  Hence, in the next commit, I will tidy it all up and use these commands:
```
pip install -e ."[full]"
pip install -e ."[test]"
```

Thanks!

File: CONTRIBUTING.md (hunk 3 above)
Comment (alexsherstinsky) [reply CONTRIBUTING.md:75]: @w4nderlust Yes -- per my previous reply, I have finally realized what was going on -- you are correct (and I will fix the relevant parts, if any, in the next commit).  Thanks.

File: CONTRIBUTING.md (hunk 1 above)
Comment (alexsherstinsky) [reply CONTRIBUTING.md:54]: @w4nderlust Update -- I checked; not a typo -- continuing the ordered was exactly my intention.  I will check to make sure that things render correctly.  Thanks.

**Comment by alexsherstinsky** [comment]
@w4nderlust After reviewing the different contributor pages, I realized that what I was going for was the right goal -- making the contributor guide clear (in terms of installation); however, my approach was ineffective and incorrect.  Here is the situation highlighting the problem:
* We have the page "https://ludwig.ai/latest/developer_guide/contributing/", which is incomplete in terms of the installation instructions; one following it would only install packages from "requirements.txt" -- the instructions for "installing [full]" are not included.  Given that this is hosted on "ludwig.ai", how do we change it to make it consistent with "https://github.com/ludwig-ai/ludwig/blob/master/CONTRIBUTING.md" (which is the focus of this PR)?  Or is the former sourcing the content from the latter automatically?
* For its part, currently the content on "https://github.com/ludwig-ai/ludwig/blob/master/CONTRIBUTING.md" has slightly different wording, but suffers from the same deficiencies as the content on "https://ludwig.ai/latest/developer_guide/contributing/" -- the lack of instructions for full, developer-ready installations.
* The content that reveals to the developer/contributor that hints about the different `extras_require` that they would be interested in appears on "https://ludwig.ai/latest/getting_started/installation/" (which I cannot change, since it is hosted on "ludwig.ai").

Hence, for the immediate term:
* I changed my suggested modification to the "https://github.com/ludwig-ai/ludwig/blob/master/CONTRIBUTING.md" page in this PR that correctly reflects the availability of the `extras_require` dependencies in `setup.py`, just like you pointed out.
* I created a task in "Linear" ("https://linear.app/predibase/issue/MLX-1516/make-contributor-guide-content-consistent-across-ludwig-sites") to make "https://ludwig.ai/latest/developer_guide/contributing/" consistent (or, perhaps, identical) with this content.

Please re-review.
Thank you.

**Comment by justinxzhao** [comment]
> * I created a task in "Linear" to make "https://ludwig.ai/latest/developer_guide/contributing/" consistent (or, perhaps, identical) with this content.

Hi @alexsherstinsky,

FYI, the contributing page in this repo and the [corresponding page](https://github.com/ludwig-ai/ludwig-docs/blob/master/docs/developer_guide/contributing.md) in ludwig docs is manually kept in sync. If there's an easy way to automate them being in sync (maybe ludwig docs loads it from [raw](https://raw.githubusercontent.com/ludwig-ai/ludwig-docs/master/docs/developer_guide/contributing.md)?), that would be great.

**Review by justinxzhao — APPROVED** [review]

**Comment by alexsherstinsky** [comment]
Thank you, @justinxzhao !  I updated the corresponding task in Linear (to explore the idea of loading from "raw").

**Pull request merged**